src/custom/utils/operator.ts

```diff
@@ -0,0 +1,131 @@
+import { ChainId } from '@uniswap/sdk'
+import { OrderCreation, ORDER_KIND_SELL } from 'utils/signatures'
+
+/**
+ * See Swagger documentation:
+ *    https://protocol-rinkeby.dev.gnosisdev.com/api/
+ */
+const API_BASE_URL: Partial<Record<ChainId, string>> = {
+  [ChainId.MAINNET]: 'https://protocol.gnosis.io/api/v1',
+  [ChainId.RINKEBY]: 'https://protocol-rinkeby.dev.gnosisdev.com/api/v1'
+  // [ChainId.xDAI]: 'https://protocol-xdai.dev.gnosisdev.com/api/v2'
+}
+
+const DEFAULT_HEADERS = {
+  headers: {
+    'Content-Type': 'application/json'
+    // TODO: Maybe add a custom header for the AppId (same as the signing tx)
+  }
+}
+
+const POST_HEADERS = {
+  method: 'POST',
+  ...DEFAULT_HEADERS
+}
+
+/**
+ * Unique identifier for the order, calculated by keccak256(orderDigest, ownerAddress, validTo),
+   where orderDigest = keccak256(orderStruct). bytes32.
+ */
+export type OrderID = string
+
+export interface OrderPostError {
+  errorType: 'MissingOrderData' | 'InvalidSignature' | 'DuplicateOrder' | 'InsufficientFunds'
+  description: string
+}
+
+function _getApiBaseUrl(chainId: ChainId): string {
+  const baseUrl = API_BASE_URL[chainId]
+
+  if (!baseUrl) {
+    throw new Error('Unsupported Network. The operator API is not deployed in the Network ' + chainId)
+  } else {
+    return baseUrl
+  }
+}
+
+async function _getErrorForBadPostOrderRequest(response: Response): Promise<string> {
+  let errorMessage: string
+  try {
+    const orderPostError: OrderPostError = await response.json()
+
+    switch (orderPostError.errorType) {
+      case 'DuplicateOrder':
+        errorMessage = 'There was another identical order already submitted'
+        break
+
+      case 'InsufficientFunds':
+        errorMessage = "The account doesn't have enough funds"
+        break
+
+      case 'InvalidSignature':
+        errorMessage = 'The order signature is invalid'
+        break
+
+      case 'MissingOrderData':
+        errorMessage = 'The order has missing information'
+        break
+
+      default:
+        console.error('Unknown reason for bad order submission', orderPostError)
+        errorMessage = orderPostError.description
+        break
+    }
+  } catch (error) {
+    console.error('Error handling a 400 error. Probably could not deserialize the JSON response')
```

Comment:
```suggestion
    console.error('Error handling a 400 error. Likely a problem deserialising the JSON response')
```

Reply:
Are we getting JSON back? I though it would be plain text with UUID on success

Reply:
it can have different things depending on the status code

201 will return a JSON --> string
404 an JSON ---> error object. It's in Alex PR for the API